assets/shaders/terrain.wgsl

```diff
@@ -146,6 +159,22 @@ fn draw_circles(base: vec4<f32>, uv: vec2<f32>) -> vec4<f32> {
     return output_color;
 }
 
+fn draw_rectangles(base: vec4<f32>, uv: vec2<f32>) -> vec4<f32> {
+    var paint = false;
+    for (var i = 0u; i < rectangles.count; i++) {
+        let rectangle = rectangles.items[i];
+        let local_uv = (rectangle.inverse_transform * vec3(uv, 1.0)).xy;
+        if all(abs(local_uv) <= rectangle.half_size + SHAPE_THICKNESS) && any(rectangle.half_size <= abs(local_uv)) {
+            paint = true;
```

Comment:
Hmm, I didn't like this logic too much either. Would just like to check, does that kind of conditional return affect the shader's performance?

Reply:
I don't know what it compiles to, but I would hope that we can spare a few iterations inside the lines. But since the lines take only a small fraction of the screen space, it will have a negligible impact.

Reply:
Roger